### PR TITLE
Floodgate playernaming fix

### LINKED ISSUE #383: Geyser Bedrock * player name compatiblity
My server have geyser and floodgate so its a crossplay server, when a player its playing from bedrock he cant create a shop because it says "Player not found".

Players from bedrock have the prefix "*" in their names, i tried changing it and removing it, and didn't work.

Please make it compatible or give me a solution.

### PR DESCRIPTION
**Pull request by MagnoBooter**
Creates a new config option VALID_PLAYERNAME_REGEXP, holding a regexp that the playername must match.
Fixes #383.

### CODE REVIEW

**Comment by Phoenix616** [comment]
I feel like `^\*?\\w+$` would be a better regex to suggest to match the asterisks. (as it only matches one or no asterisks compared to the 0-unlimited ones from your example one)

Also I don't think this has reloading support right now (in the InvalidNameIgnorer). Do you think it could be possible to add reload support there? (Especially as the ChestShopSign method does support it)

File: src/main/java/com/Acrobot/ChestShop/Signs/ChestShopSign.java [hunk 1]
```diff
@@ -149,10 +148,17 @@ public static boolean isOwner(Player player, Sign sign) {
     }
 
     public static boolean isValidPreparedSign(String[] lines) {
-        for (int i = 0; i < 4; i++) {
+    	if ((!isAdminShop(lines[0])) && (lines[0].length() > 0)) {
+    		Pattern PlayernamePattern = Pattern.compile(Properties.VALID_PLAYERNAME_REGEXP);
+    		if (!PlayernamePattern.matcher(lines[0]).matches()) {
+    			return false;
+    		}
+    	}
```
Comment (Phoenix616) [inline src/main/java/com/Acrobot/ChestShop/Signs/ChestShopSign.java:156]: Please use four spaces instead of tabs, also camelCase for the variables.

**Comment by Phoenix616** [comment]
I found the time to look at this and I noticed an issue with using the same regex for both checks: The InvalidNameIgnorer is only supposed to validate the actual client name on transactions while the `ChestShopSign` patterns also need to match at least the `:` character in order for it to be possible for player IDs to be used in the case of duplicate short names. (e.g. `ShortName:4`) which is why that was included in the original regex. (not sure why the other values are there, maybe just to match possible admin shop names?)

I'm not sure what the best solution is, including those characters in the regex isn't ideal as that would mean that actually invalid player names will pass through the `InvalidNameIgnorer` check...

**Comment by Camotoy** [comment]
Just a heads-up that Floodgate is very likely transitioning to `.` as the default prefix, as that works better in many ways while still being a unique character that is not available to normal Java players. If documentation about Floodgate is included in ChestShop's config, I would phrase it in a way that specifies whatever Floodgate's prefix is and not just `*` (as that can be changed).

**Pull request merged**